test/unit/src/common/components/beta-notification/t_beta-notification.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import expect from 'expect';
+import { shallow } from 'enzyme';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+
+import { BetaNotificationView } from '../../../../../../src/common/components/beta-notification';
+
+// importing BetaNotificationTriggerView through proxyquire to stub localforage
+import { makeLocalForageStub } from '../../../../../helpers';
+const localForageStub = makeLocalForageStub();
+const BetaNotificationTriggerView = proxyquire.noCallThru().load(
+  '../../../../../../src/common/components/beta-notification/trigger',
+  { 'localforage': localForageStub }
+).BetaNotificationTriggerView;
+
+describe('<BetaNotificationView />', () => {
+  it('should not render if notification visibility is set to false', () => {
+    const wrapper = shallow(<BetaNotificationView isVisible={false} />);
+    expect(wrapper.type()).toBe(null);
+  });
+
+  it('should not render Notification visibility is set to true', () => {
+    const wrapper = shallow(<BetaNotificationView isVisible={true} />);
+    expect(wrapper.is('Notification')).toBe(true);
+  });
+});
+
+
+describe('<BetaNotificationTriggerView />', () => {
+  let clock;
+
+  before(() => {
+    clock = sinon.useFakeTimers();
+  });
+
+  after(() => {
+    clock.restore();
+  });
+
+  context('when user is logged in', () => {
+    let wrapper;
+    let callback;
+
+    beforeEach(async () => {
+      callback = expect.createSpy();
+      wrapper = shallow(<BetaNotificationTriggerView authenticated={true} showNotification={callback}/>);
+
+      // componentDidMount is async because of internal localforage usage
+      // and because mocked localforage uses setTimeout to call async callbacks
+      // we need to first store reference to componentDidMount promise:
+      const componentDidMount = wrapper.instance().componentDidMount();
+      // and then tick the mocked clock to make localforage resolve properly
+      clock.tick(1);
+      // and finally we can wait for componentDidMount to fully resolve
+      await componentDidMount;
```

Comment:
This was a much harder beast to test than I thought it would.

We are using mocked setTimeout (from `clock`), but internally used `localforage` stub also uses `setTimeout` to call its callbacks asynchronously. So localforage promises don't resolve until we tick the clock. And only then we can wait till async componentDidMount resolves...